internal/provider/vaultradar/integration_connection.go

```diff
@@ -0,0 +1,199 @@
+	SetName(types.String)
+	GetAuthKey() (string, diag.Diagnostics)
+	GetDetails() (string, diag.Diagnostics)
+	SetDetails(string) diag.Diagnostics
+}
+
+func (r *integrationConnectionResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + r.TypeName
+}
+
+func (r *integrationConnectionResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = r.ConnectionSchema
+}
+
+func (r *integrationConnectionResource) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+	client, ok := req.ProviderData.(*clients.Client)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *clients.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+		return
+	}
+	r.client = client
+}
+
+func (r *integrationConnectionResource) ModifyPlan(ctx context.Context, req resource.ModifyPlanRequest, resp *resource.ModifyPlanResponse) {
+	modifiers.ModifyPlanForDefaultProjectChange(ctx, r.client.Config.ProjectID, req.State, req.Config, req.Plan, resp)
+}
+
+func (r *integrationConnectionResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	plan, diags := r.GetPlan(ctx, req.Plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	projectID := r.client.Config.ProjectID
+	if !plan.GetProjectID().IsUnknown() {
+		projectID = plan.GetProjectID().ValueString()
+	}
+
+	errSummary := "Error creating Radar Integration Connection"
+
+	// Check for an existing connection with the same name.
+	existing, err := clients.GetIntegrationConnectionByName(ctx, r.client, projectID, plan.GetName().ValueString())
+	if err != nil && !clients.IsResponseCodeNotFound(err) {
+		resp.Diagnostics.AddError(errSummary, err.Error())
+	}
+	if existing != nil {
+		resp.Diagnostics.AddError(errSummary, fmt.Sprintf("Connection with name: %q already exists.", plan.GetName().ValueString()))
+		return
+	}
+
+	authKey, diags := plan.GetAuthKey()
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	details, diags := plan.GetDetails()
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	res, err := clients.CreateIntegrationConnection(ctx, r.client, projectID, service.CreateIntegrationConnectionBody{
+		IntegrationType: r.IntegrationType,
+		IsSink:          true,
+		IsSource:        false,
+		Name:            plan.GetName().ValueString(),
+		AuthKey:         authKey,
+		Details:         details,
+	})
+	if err != nil {
+		resp.Diagnostics.AddError(errSummary, err.Error())
+		return
+	}
+
+	plan.SetID(types.StringValue(res.GetPayload().ID))
+	plan.SetProjectID(types.StringValue(projectID))
+	resp.Diagnostics.Append(resp.State.Set(ctx, plan)...)
```

Comment:
nit: Can we add trace logs similar to the Radar resource?
https://github.com/hashicorp/terraform-provider-hcp/blob/cd7cff2c530a9c8d65d26f41ebf659a23501e056/internal/provider/vaultradar/resource_radar_source_github_enterprise.go#L149

Reply:
Actually I was planning on removing those. I had initially put them there just so I could confirm what was going on, but there is enough in the logs that make these log unnecessary for a success state.